GSA/digitalgov.gov
# HCD #3: Refactor Side Navigation

## Summary

Refactor the side navigation component to work with multiple "levels" of navigation and display in page navigation links.

### Preview

[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jtm-refactor-sidenav/)



### Solution
<img width="1439" alt="Screenshot 2023-08-04 at 4 06 34 PM" src="https://github.com/GSA/digitalgov.gov/assets/56370370/47bfbff0-03e2-46c9-a339-18d26ebc24ae">

### How To Test

1. Set the `showInPageNav` parameter in a `guidenav` file to `true`
2. Verify that the navigation displays and functions correctly for pages with no sub-sections
3. Verify that the navigation displays and functions correctly for pages with multiple sub-sections (This will require adding sub-sections to a `guidenav` file. Check out the [HCD guides](https://github.com/GSA/digitalgov.gov/blob/jtm-create-hcd-guides/data/guidenav/hcd-design-concepts.yml) for ways to do this.)

> [!WARNING]  
> There is a [known bug in Chromium browsers](https://bugs.chromium.org/p/chromium/issues/detail?id=1043933) that affects scrolling while another element is scrolling. This means that we are unable to smoothly scroll to in page navigation links when the menu bar has not yet stuck to the top of the page (as the menu bar automatically scrolls when it sticks to the top of the page). There is a workaround for this on line 58 of `guide-sidenav.js`, but it should be removed once the Chromium bug is resolved.

---

### Dev Checklist
